apps/ejabberd/src/ejabberd_hooks.erl

```diff
@@ -242,3 +250,20 @@ run_fold1([{_Seq, Module, Function} | Ls], Hook, Val, Args) ->
         NewVal ->
             run_fold1(Ls, Hook, NewVal, Args)
     end.
+
+hook_apply_function(_Module, Function, _Hook, Val, Args) when is_function(Function) ->
+    safely:apply(Function, [Val | Args]);
+hook_apply_function(Module, Function, Hook, Val, Args) ->
+    record(Hook, Module, Function, safely:apply(Module, Function, [Val | Args])).
```

Comment:
Assigning result from `safe:apply` to its own var in its own line may improve readability a bit.